### DISCUSSION TITLE
Acaba l'sprint 2 i comença l'sprint 3.

### DISCUSSION THREAD

**Discussion by FherStk** [ElPuig/IMS#6] [1. IMS: general]
Bon dia a tothom,

Perdoneu el retard (de nou), però l'sprint 2 (temps per a proposar els models de dades i acordar funcionalitats) va acabar el dia 31 i el dia 1 va començar l'sprint 3. Com ja sabeu, hem definit els sprints perquè tinguin una durada d'un mes i durant aquest tercer sprint ens toca el següent:

1. Prepararé un esquelet de mòdul que inclogui els models de dades pactats, els permisos inicials i que inclogui un script que permeti instal·lar-lo dins Odoo sense haver d'executar les comandes necessàries (pot resultar tediós en el moment de fer proves).
2. Un cop estigui completat pujaré l'esquelet al "main" perquè tots tinguem un codi inicial.
3. Bloquejaré la branca principal (main) per a evitar commits accidentals. Ja parlarem més endavant de com gestionem els canvis al "main" però crec que queda temps-. 
4. Haureu de fer una branca amb el vostre nom, a partir del "main" per a poder treballar i fer commit & push sense generar conflictes amb la resta de companys. 

Salut i gràcies a tots,
Fer

**Comment by FherStk** [comment]
Bones,

Ja he fet el primer push amb un esquelet molt bàsic de l'IMS, però hauria de funcionar als vostres sistemes. És important que doneu permisos a la carpeta de desenvolupament (https://github.com/ElPuig/IMS/wiki/2.3.-Creaci%C3%B3-de-m%C3%B2duls-per-a-Odoo#permisos) i després només caldria:

1. Clonar el projecte dins la carpeta "myModules" (o la que hagueu creat): `git clone https://github.com/ElPuig/IMS.git ims`
2. Entrar dins la carpeta: `cd ims`
3. Instal·lar el mòdul amb `./install.sh`

El mòdul hauria d'aparèixer instal·lat dins Odoo i ser operatiu, qualsevol canvi al codi font es pot instal·lar amb './update.sh'.

Salut i bones vacances,
Fer

**Comment by jherna** [comment]
Bon pasqua,

Tinc alguns dubtes, després de fer l'install.sh i al entrar al odoo, em 
demana que trii la base de dades. En la meva instal·lació inicial vaig 
donar-li el nom de "ims" i ara en tinc una altra que es diu "odoo" amb 
l'execució de l'script. Si entro a la original, ims, puc entrar sense 
problemes amb el password que li vaig dir, però per entrar a la "odoo", 
no tinc password.

Alguna idea?

Jordi

El 4/4/23 a les 20:23, Fher ha escrit:
>
> Bones,
>
> Ja he fet el primer push amb un esquelet molt bàsic de l'IMS, però 
> hauria de funcionar als vostres sistemes. És important que doneu 
> permisos a la carpeta de desenvolupament 
> (https://github.com/ElPuig/IMS/wiki/2.3.-Creaci%C3%B3-de-m%C3%B2duls-per-a-Odoo#permisos) 
> i després només caldria:
>
>  1. Clonar el projecte dins la carpeta "myModules" (o la que hagueu
>     creat): |git clone https://github.com/ElPuig/IMS.git|
>  2. Entrar dins la carpeta: |cd ims|
>  3. Instal·lar el mòdul amb |./install.sh|
>
> El mòdul hauria d'aparèixer instal·lat dins Odoo i ser operatiu, 
> qualsevol canvi al codi font es pot instal·lar amb './update.sh'.
>
> Salut i bones vacances,
> Fer
>
> —
> Reply to this email directly, view it on GitHub 
> <https://github.com/ElPuig/IMS/discussions/6#discussioncomment-5524245>, 
> or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/AAUAHLZA6S3X7M64OQUY433W7RRJBANCNFSM6AAAAAAWSMHUWQ>.
> You are receiving this because you are subscribed to this 
> thread.Message ID: 
> ***@***.***>
>
-- 
Jordi Hernàndez
Tutor de 2n de DAM - INS Puig Castellar

**Reply by jherna** [reply]
Solucionat el tema per entrar a la base de dades que m'ha creat l'script. Amb admin:admin (user:password) ha anat bé!
Per altra banda al activar el mòdul IMS han aparegut els següents errors:

> Exception: Module loading IMS failed: file IMS/security/ir.model.access.csv could not be processed:
 No matching record found for external id 'ims.model_ims_mp' in field 'Model'
No matching record found for external id 'ims.model_ims_uf' in field 'Model'
Missing required value for the field 'Model' (model_id)
Missing required value for the field 'Model' (model_id)

**Reply by jherna** [reply]
Aquest és un altre error, ara però dins la base de dades ims, després de modificar l'script. 

> Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/odoo/addons/base/models/ir_fields.py", line 118, in fn
    converted[field], ws = converters[field](value)
  File "/usr/lib/python3/dist-packages/odoo/addons/base/models/ir_fields.py", line 561, in _str_to_many2one
    id, _, w2 = self.db_id_for(model, field, subfield, record[subfield])
  File "/usr/lib/python3/dist-packages/odoo/addons/base/models/ir_fields.py", line 499, in db_id_for
    raise self._format_import_error(
ValueError: ("No s'han trobat registres coincidents per id extern 'ims.model_ims_mp' al camp '%(camp)s'", {'moreinfo': {'name': 'Possible Values', 'type': 'ir.actions.act_window', 'target': 'new', 'view_mode': 'tree,form', 'views': [(False, 'list'), (False, 'form')], 'context': {'create': False}, 'help': 'Veure tots els possibles valors', 'res_model': 'ir.model.data', 'domain': [('model', '=', 'ir.model')]}})

**Reply by FherStk** [reply]
Hola Jordi,

He vist que en fer el `git clone` la carpeta IMS que descarrega està en majúscules i ha d'estar en minúscules. He tornat a fer el procés en una màquina nova i m'ha funcionat d'aquesta manera. El que no he provat és de canviar-li el nom a la BBDD, però això només hauria d'influir en l'script d'instal·lació, que el canviaré perquè accepti la BBDD com a paràmetre opcional. 

Pots provar si és un problema amb la carpeta?

Salut,
Fer

**Comment by FherStk** [comment]
Perdoneu, veig que el mòdul no permet crear nous elements, suposo que és un problema de permisos (fitxer `security/ir.model.access.csv`). M'ho miro quan tingui un momentet.

**Comment by jherna** [comment]
Perfecte, eren les majúscules de la carpeta.

Merci

El 10/4/23 a les 11:22, Fher ha escrit:
>
> Hola Jordi,
>
> He vist que en fer el |git clone| la carpeta IMS que descarrega està 
> en majúscules i ha d'estar en minúscules. He tornat a fer el procés en 
> una màquina nova i m'ha funcionat d'aquesta manera. El que no he 
> provat és de canviar-li el nom a la BBDD, però això només hauria 
> d'influir en l'script d'instal·lació, que el canviaré perquè accepti 
> la BBDD com a paràmetre opcional.
>
> Pots provar si és un problema amb la carpeta?
>
> Salut,
> Fer
>
> —
> Reply to this email directly, view it on GitHub 
> <https://github.com/ElPuig/IMS/discussions/6#discussioncomment-5569180>, 
> or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/AAUAHL7OJYZGLZLMOXDOMVTXAPGMJANCNFSM6AAAAAAWSMHUWQ>.
> You are receiving this because you commented.Message ID: 
> ***@***.***>
>
-- 
Jordi Hernàndez
Tutor de 2n de DAM - INS Puig Castellar
